materialize-slack/main.go

```diff
@@ -0,0 +1,357 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+
+	schemagen "github.com/estuary/connectors/go/schema-gen"
+	boilerplate "github.com/estuary/connectors/materialize-boilerplate"
+	"github.com/estuary/flow/go/protocols/fdb/tuple"
+	pf "github.com/estuary/flow/go/protocols/flow"
+	pm "github.com/estuary/flow/go/protocols/materialize"
+	log "github.com/sirupsen/logrus"
+	"github.com/slack-go/slack"
+	"go.gazette.dev/core/consumer/protocol"
+)
+
+// driver implements the pm.DriverServer interface.
+type driver struct{}
+
+type config struct {
+	Credentials CredentialConfig `json:"credentials" jsonschema:"required,title=Authentication" jsonschema_extras:"x-oauth2-provider=slack"`
+}
+
+// Validate returns an error if the config is not well-formed.
+func (c config) Validate() error {
+	if err := c.Credentials.validateClientCreds(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c config) buildAPI() (*SlackAPI, error) {
+	var api = c.Credentials.SlackAPI()
+	if err := api.AuthTest(); err != nil {
+		return nil, fmt.Errorf("error verifying authentication: %w", err)
+	}
+	return api, nil
+}
+
+type resource struct {
+	Channel      string            `json:"channel" jsonschema:"title=Channel,description=The name of the channel to post messages to (or a raw channel ID like \"id:C123456\")"`
+	SenderConfig SlackSenderConfig `json:"sender_config" jsonschema:"title=Configure Appearance"`
+}
+
+func (r resource) Validate() error {
+	if r.Channel == "" {
+		return fmt.Errorf("missing required channel name/id")
+	}
+	return nil
+}
+
+type driverCheckpoint struct {
+	BindingCollections []string `json:"binding_collections"`
+}
+
+func (c driverCheckpoint) Validate() error {
+	return nil
+}
+
+func (driver) Spec(ctx context.Context, req *pm.Request_Spec) (*pm.Response_Spec, error) {
+	log.Debug("handling Spec request")
```

Comment:
These `log.Debug` statements at the start of every driver/transactor method don't seem like they will be very useful, and also generally redundant with debug messages from the runtime. Can we remove them?

Reply:
Removed all of them this time I'm pretty sure 😅